core/src/main/java/greencity/controller/HabitCommentController.java

```diff
@@ -0,0 +1,271 @@
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Size;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+@Validated
+@AllArgsConstructor
+@RestController
+@RequestMapping("/habits")
+public class HabitCommentController {
+    private final CommentService commentService;
+
+    /**
+     * Method for creating {@link CommentVO}.
+     *
+     * @param habitId id of {@link HabitVO} to add comment to.
+     * @param request dto for {@link CommentVO} entity.
+     * @return dto {@link AddCommentDtoResponse}
+     */
+    @Operation(summary = "Add comment")
+    @ResponseStatus(value = HttpStatus.CREATED)
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "201", description = HttpStatuses.CREATED,
+            content = @Content(schema = @Schema(implementation = AddCommentDtoResponse.class))),
+        @ApiResponse(responseCode = "400", description = HttpStatuses.BAD_REQUEST,
+            content = @Content(examples = @ExampleObject(HttpStatuses.BAD_REQUEST))),
+        @ApiResponse(responseCode = "401", description = HttpStatuses.UNAUTHORIZED,
+            content = @Content(examples = @ExampleObject(HttpStatuses.UNAUTHORIZED))),
+        @ApiResponse(responseCode = "404", description = HttpStatuses.NOT_FOUND,
+            content = @Content(examples = @ExampleObject(HttpStatuses.NOT_FOUND))),
+    })
+    @PostMapping("/{habitId}/comments")
+    public ResponseEntity<AddCommentDtoResponse> save(@PathVariable Long habitId,
+        @Valid @RequestBody AddCommentDtoRequest request,
+        @Parameter(hidden = true) @CurrentUser UserVO userVO) {
+        return ResponseEntity
+            .status(HttpStatus.CREATED)
+            .body(commentService.save(ArticleType.HABIT, habitId, request, userVO));
+    }
+
+    /**
+     * Method to get certain comment for {@link HabitVO} specified by commentId.
+     *
+     * @param id specifies {@link CommentDto} to which we search for comments
+     * @return comment to certain habit specified by commentId.
+     */
+    @Operation(summary = "Get comment by id")
+    @ResponseStatus(value = HttpStatus.OK)
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = HttpStatuses.OK,
+            content = @Content(schema = @Schema(implementation = AddCommentDtoResponse.class))),
+        @ApiResponse(responseCode = "400", description = HttpStatuses.BAD_REQUEST,
+            content = @Content(examples = @ExampleObject(HttpStatuses.BAD_REQUEST))),
+        @ApiResponse(responseCode = "401", description = HttpStatuses.UNAUTHORIZED,
```

Comment:
based on the configs this endpoint is available for all users (you use `permitAll()` for this endpoint)
Please check for all added endpoint if you use correct configs and update response codes